cmd2.py

```diff
@@ -547,6 +547,230 @@ def strip_ansi(text):
+            if it's a string, it should contain one or more of:
+                {super_prompt}:     The current cmd's prompt
+                {command}:          The command in the current cmd with which it was called
+                {sub_prompt}:       The subordinate cmd's original prompt
+            the default is "{super_prompt}{command} {sub_prompt}"
+
+        shared_attributes (dict):   dict of the form {'subordinate_attr': 'parent_attr'}
+            the attributes are copied to the submenu at the last moment; the submenu's
+            attributes are backed up before this and restored afterward
+
+        require_predefined_shares: The shared attributes above must be independently
+            defined in the subordinate Cmd (default: True)
+
+        create_subclass: put the modifications in a subclass rather than modifying
+            the existing class (default: False)
+        """
+        self.submenu = submenu
+        self.command = command
+        self.aliases = aliases
+
+        if reformat_prompt is not None and not isinstance(reformat_prompt, str):
+            raise ValueError("reformat_prompt should be either a format string or None")
+        self.reformat_prompt = reformat_prompt
+
+        if require_predefined_shares:
+            for attr in shared_attributes.keys():
+                if not hasattr(submenu, attr):
+                    raise AttributeError("The shared attribute '{attr}' is not defined in {cmd}. Either define {attr} "
+                                         "in {cmd} or set require_predefined_shares=False."
+                                         .format(cmd=submenu.__class__.__name__, attr=attr))
+
+        self.shared_attributes = {} if shared_attributes is None else shared_attributes
+        self.create_subclass = create_subclass
+
+    def __call__(self, cmd_obj):
+        """Creates a subclass of Cmd wherein the given submenu can be accessed via the given command"""
+        def enter_submenu(parent_cmd, line):
+            """
+            This function will be bound to do_<submenu> and will change the scope of the CLI to that of the
+            submenu.
+            """
+            submenu = self.submenu
+            original_attributes = {attr: getattr(submenu, attr, AddSubmenu._Nonexistent)
+                                   for attr in self.shared_attributes.keys()
+                                   }
+            try:
+                # copy over any shared attributes
+                for sub_attr, par_attr in self.shared_attributes.items():
+                    setattr(submenu, sub_attr, getattr(parent_cmd, par_attr))
+
+                if line.parsed.args:
+                    # Remove the menu argument and execute the command in the submenu
+                    line = submenu.parser_manager.parsed(line.parsed.args)
+                    submenu.precmd(line)
+                    ret = submenu.onecmd(line)
+                    submenu.postcmd(ret, line)
+                else:
```

Comment:
I'd like to try to find a unit/integration test which covers this **else** case to help increase code coverage.  Can you suggest one?